src/foam/core/CompoundAgency.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+import java.util.ArrayList;
+
+/**
+ * A ContextAgent which is itself an Agency.
+ * When ContextAgents are submit()'ed to the service it queues them
+ * but doesn't execute them until it is itself executed.
+ **/
+public class CompoundAgency
```

Comment:
Why is this class needed? Why can't CompoundContextAgent just implement Agency directly?